src/compiler/compile/render_dom/wrappers/shared/mark_each_block_bindings.ts

```diff
@@ -10,7 +11,13 @@ export default function mark_each_block_bindings(
 	// we need to ensure that the each block creates a context including
 	// the list and the index, if they're not otherwise referenced
 	binding.expression.references.forEach(name => {
-		const each_block = parent.node.scope.get_owner(name);
+		const each_block = parent.node.scope.get_owner(name) as EachBlock;
+		if (each_block) {
+			const rest_node = each_block.context_rest_properties.get(name);
+			if (rest_node) {
+				parent.renderer.component.warn(rest_node as any, compiler_warnings.invalid_rest_eachblock_binding(name));
+			}
+		}
 		if (each_block) {
 			(each_block as EachBlock).has_binding = true;
 		}
```

Comment:
It can be merged into the above if statement after  commit my above suggestion.

Reply:
Resolved